libs/optimizer/src/lib/optimizer.ts

```diff
@@ -0,0 +1,348 @@
+import { Solution, solver } from 'igc-xc-score';
+import { BRecord, IGCFile } from 'igc-parser';
+import { createSegments } from './utils/createSegments';
+import { concatTracks } from './utils/concatTracks';
+import { ScoringRules, scoringRules } from './scoringRules';
+import { getDistance } from 'geolib';
+
+// When the track has not enough points (<5), we build a new one by adding interpolated points between existing ones.
+// see this issue https://github.com/mmomtchev/igc-xc-score/issues/231 for removing this limitation
+const MIN_POINTS = 5;
+const NUM_SEGMENTS_BETWEEN_POINTS = 2;
+
+// For adding interpolated points, this constant adjusts the proximity of the points to the starting point of
+// the segment. We want the added points to be very close to the starting points of the segment so that the solution
+// points returned by the solver are as close as possible (or may be equal) to one of the original points of the track.
+const DISTRIBUTION_FACTOR_FOR_ADDED_POINTS = 1e-5;
+
+
+export interface LatLonAltTime {
+  alt: number;
+  lat: number;
+  lon: number;
+  /**
+   * time in seconds elapsed since the beginning of the track (see ScoringTrack.minTimeSec)
+   */
+  timeSec: number;
+}
+
+export interface ScoringTrack {
+  /**
+   * the points that describe the track
+   */
+  points: LatLonAltTime[];
+  /**
+   * the instant when the track starts (seconds elapsed since 01-01-1970T00:00:00.000)
+   * the "timeSec" values in LatLonAltTime's are offsets according to this instant.
+   */
+  startTimeSec: number;
+}
+
+export interface OptimizationOptions {
+  /**
+   * maximum duration in milliseconds for an optimization round trip.
+   * If undefined, calculation duration is unbounded.
+   */
+  maxCycleDurationMs?: number;
+  /**
+   * maximum number of iterations allowed for an optimization round trip.
+   * If undefined, number of allowed iterations is unbounded
+   */
+  maxNumCycles?: number;
+}
+
+/**
+ * optimize function argument
+ */
+export interface OptimizationRequest {
+  /**
+   * the ScoringTrack to optimize
```

Comment:
nit: You can drop those kind of comments, they don't add value

Reply:
Ok. Removed comments without value